c++/src/kj/refcount.h

```diff
@@ -115,6 +142,106 @@ Own<T> Refcounted::addRefInternal(T* object) {
   return Own<T>(object, *refcounted);
 }
 
+template <typename T>
+Rc<T> Refcounted::addRcRefInternal(T* object) {
+  Refcounted* refcounted = object;
+  ++refcounted->refcount;
+  return Rc<T>(object);
+}
+
+template<typename T>
+class Rc: private Own<T> {
+  // Smart pointer for reference counted objects. 
+  //
+  // There are only two ways to obtain new Rc instances:
+  // - use kj::rc<T>(...) function to create new T.
+  // - use kj::Rc::addRef() and the existing Rc instance.
+  // - use EnableAddRefToThis to allow T instance to add new references to itself.
+  //
+  // kj::Rc<T> can be used everywhere Own<T> is expected.
```

Comment:
Can it? Doesn't the private inheritance block that?

Reply:
well, initially it was public, but James has convinced me to drop it. Let me cleanup.